posts/python-in-production-part-5-of-5.md

```diff
@@ -0,0 +1,250 @@
+
+## Pipeline Plugin
+
+When Jenkins pulls our source code, we want to be able to tell it to look for our Jenkinsfile.  This will allow us to update our build pipeline in our source code, without having to edit the jenkins job.  Search for `Pipeline` in the plugin manager.  Now you are ready to press install.
+
+# Create the Jenkins Job
+
+To create a Jenkins Job, click the `New Item` button on the left side bar. You should see the option `Pipeline`.  If you do not see this option, you may need to restart Jenkins.
+
+Once you have named your job and gotten into the configuration screen, you will see there are many fields and options.  For Description, I wrote...
+```text
+Build our sample-python-project.
+```
+
+Beneath the description are some checkboxes.  The one we are interested in is `GitHub project`.  Click the box and populate your github url for the sample-python-project repository.  You can skip the build triggers section for now, but this is where you would come if you wanted to trigger a build via some type of automated action.
+
+## Pipeline setup
+
+Further down you will see a section called `Pipeline`.  Select the option under `Definition` that says ***Pipeline script from SCM***.  SCM just means source control management.  This is telling Jenkins that we want it to use our Jenkins file in our GitHub Project Repo to control the build.
+
+Choose Git under SCM, fill out your Repository URL, skip the Credential section, because this is a public repo, and then specify the branch to build.  Branch is typically `*/master` or `*/main`.   Leave the rest of the options as they are and press **save**.
+
+# Run the Build Job
+
+Now that you have created your Jenkins job, which is a pipeline job, we are ready to run it!  Navigate back to the main dashboard and click on your Jenkins Job.  On the Job screen, you will see some options on the left.  Click `Build Now`.  Further down on the left side bar, you should see your job pop up in the queue.  If you click on the job, you can navigate to the `Console Output` screen.  This will show you the `stdout` of your build job.
+
+If the job passes, you will see `Finished: SUCCESS` at the bottom of the console output.  Chances are, you may need to iterate over you build script and Jenkins setup a few times before you get a successful build.  Let's look at some troubleshooting steps.
+
+## Build Environment
+
+Jenkins runs in a linux vm.  It can be ran directly on a machine or inside a dockerfile.  Because of this reason, it is constrained by the machine it is running on.  During the build process, you may have gotten an error that Python3 was not installed.  In order to get around this, we need to add python to the build machine.  
```

Comment:
```suggestion
Jenkins runs in a linux vm.  It can be run directly on a machine or inside a dockerfile.  Therefore, it is constrained by the machine it is running on.  During the build process, you may have gotten an error that Python3 was not installed.  To get around this, we need to add python to the build machine.  
```

Reply:
Standardize linux and vm as well. I have Grammarly on chrome and I think it automatically capitalized both for me earlier.